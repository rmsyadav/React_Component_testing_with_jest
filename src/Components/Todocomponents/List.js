import axios from "axios";
import { useEffect, useState } from 'react';
import "../../Stylesheet/Todo.css";


const List =()=> {

   const [comments,setcomments] = useState([]);

   useEffect(()=>{
        axios.get('https://dummyjson.com/comments').then((res)=>{
            setcomments(res.data.comments);
        })
   },[]) 

  return (
   <>
    <div className="container mt-3">
        <h2>Numbered List Group</h2>
        </div>
        
        <div className="container mt-3">
        <ol className="list-group list-group-light">
               <li className="list-group-item column-list-text d-flex justify-content-between align-items-center">
                    <span className="badge rounded-pill text-dark">USERID</span>
                    <div className="d-flex flex-column justify-content-between align-items-center">
                      <div className="fw-bold text-dark">USERNAME/COMMENTS</div>
                    </div>
                    <span className="badge rounded-pill text-dark">POSTID</span>
                </li>
            {comments && comments.map((comment,id)=>{
                return (
                <li className="list-group-item d-flex justify-content-between align-items-center" data-testid="list-group-item" key={id}>
                     <span className="badge rounded-pill text-success">{comment.id}</span>
                    <div className="d-flex flex-column justify-content-between align-items-center">
                    <div className="fw-bold">{comment.user.username}</div>
                    <div className="text-muted">{comment.body}</div>
                    </div>
                    <span className="badge rounded-pill text-success">{comment.postId}</span>
                </li>
               )
            })}
        </ol>
   </div>
   </>
  );
}

export default List;
