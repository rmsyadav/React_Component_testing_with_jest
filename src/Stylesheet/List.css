
.spinner-icon{
    position: absolute;
    top:50%;
    left: 50%;
}
.search-filter{
    column-gap: 20px;
    flex-wrap: wrap;
}
.filter{
    column-gap: 10px;
    flex-wrap: wrap; 
}
.animate-charcter
{
   text-transform: uppercase;
  background-image: linear-gradient(
    -225deg,
    #231557 0%,
    #44107a 29%,
    #ff1361 67%,
    #fff800 100%
  );
  background-size: auto auto;
  background-clip: border-box;
  background-size: 200% auto;
  color: #fff;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  animation: textclip 2s linear infinite;
  display: inline-block;
  font-size: 20px;
}

@keyframes textclip {
  to {
    background-position: 200% center;
  }
}