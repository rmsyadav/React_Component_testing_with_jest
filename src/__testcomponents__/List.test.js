 import { rest } from 'msw';
 import renderer from 'react-test-renderer';
import { render, screen ,act} from '@testing-library/react';
import { setupServer } from 'msw/node';
import List from '../Components/Todocomponents/List';
import { Provider } from 'react-redux';
import store from '../StateContainer/Store';
const mockTasks = {
  comments:[
    {
      id: 1,
      body: "This is some awesome thinking!",
      postId: 100,
      user: {
          id: 63,
          username: "eburras1q"
      }
  },
  {
      id: 2,
      body: "What terrific math skills you’re showing!",
      postId: 27,
      user: {
          id: 71,
          username: "omarsland1y"
      }
  },
 ]
}

const getTasksPath ='https://dummyjson.com/comments';

const tasksHandler = rest.get(getTasksPath, async (req, res, ctx) =>{
   return  res(ctx.json(mockTasks))
  }  
);

const mswServer = setupServer(tasksHandler);



describe('Component: TaskList', () => {

    beforeAll(() => mswServer.listen());
    afterEach(() => mswServer.resetHandlers());
    afterAll(() => mswServer.close());
    
    it('List component snapshot testing ',()=>{

      const list = renderer.create(<Provider store={store}><List></List></Provider>).toJSON();
      expect(list).toMatchSnapshot();
    })

    it('displays returned tasks on successful fetch', async () => {
      
      
      await act(()=>{
        render(<Provider store={store}><List></List></Provider>);
      })
      const displayedTasks = await screen.findAllByTestId("list-group-item");
      expect(displayedTasks).toHaveLength(2);
      expect(screen.getByText('This is some awesome thinking!')).toBeInTheDocument();
      expect(screen.getByText('What terrific math skills you’re showing!')).toBeInTheDocument();
      
    });
  
  });