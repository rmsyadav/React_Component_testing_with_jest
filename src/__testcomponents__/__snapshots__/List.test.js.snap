// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TaskList List component snapshot testing  1`] = `
Array [
  <div
    className="container mt-3"
  >
    <h2>
      Numbered List Group
    </h2>
  </div>,
  <div
    className="container mt-3"
  >
    <ol
      className="list-group list-group-light"
    >
      <li
        className="list-group-item column-list-text d-flex justify-content-between align-items-center"
      >
        <span
          className="badge rounded-pill text-dark"
        >
          USERID
        </span>
        <div
          className="d-flex flex-column justify-content-between align-items-center"
        >
          <div
            className="fw-bold text-dark"
          >
            USERNAME/COMMENTS
          </div>
        </div>
        <span
          className="badge rounded-pill text-dark"
        >
          POSTID
        </span>
      </li>
    </ol>
  </div>,
]
`;
